### ISSUE TITLE
TypeError: StaticPartitionMapping is not JSON serializable

### ISSUE THREAD

**Issue by gustavo-delfosim** [dagster-io/dagster#11933]
### Dagster version

1.1.14

### What's the issue?

When I run `dagit` in a code containing a `StaticPartitionMapping` instance, I get the following error:
```
Error loading repository location sensor:TypeError: Object of type StaticPartitionMapping is not JSON serializable
```
Full Stack Trace:
```
File "/home/gustavo/miniconda3/lib/python3.10/site-packages/dagster/_grpc/server.py", line 512, in _get_serialized_external_repository_data
    return serialize_dagster_namedtuple(
  File "/home/gustavo/miniconda3/lib/python3.10/site-packages/dagster/_serdes/serdes.py", line 346, in serialize_dagster_namedtuple
    return _serialize_dagster_namedtuple(nt, whitelist_map=_WHITELIST_MAP, **json_kwargs)
  File "/home/gustavo/miniconda3/lib/python3.10/site-packages/dagster/_serdes/serdes.py", line 352, in _serialize_dagster_namedtuple
    return seven.json.dumps(pack_inner_value(nt, whitelist_map, _root(nt)), **json_kwargs)
  File "/home/gustavo/miniconda3/lib/python3.10/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
  File "/home/gustavo/miniconda3/lib/python3.10/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/home/gustavo/miniconda3/lib/python3.10/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/home/gustavo/miniconda3/lib/python3.10/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__}
```

### What did you expect to happen?

I expected `dagit` to properly run with the `StaticPartitionMapping`.

### How to reproduce?

Create a file `error.py`:
```
from dagster import (
    AssetIn,
    Definitions,
    StaticPartitionMapping,
    StaticPartitionsDefinition,
    asset,
)

upstream_parts = StaticPartitionsDefinition(["a1", "a2", "b1", "b2"])
downstream_parts = StaticPartitionsDefinition(["a", "b"])
mapping = StaticPartitionMapping({"a1": "a", "a2": "a", "b1": "b", "b2": "b"})


@asset(partitions_def=upstream_parts)
def x():
    return 1


@asset(
    partitions_def=upstream_parts,
    ins={"x": AssetIn(partition_mapping=mapping)},
)
def y(x):
    return x


defs = Definitions(assets=[x, y])
```

Run `dagit -f error.py` and check logs.
The error can also be seen on the dagit UI in the Deployment tab.

### Deployment type

Local

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by sryza** [comment]
I believe this was fixed here already: https://github.com/dagster-io/dagster/pull/11738#pullrequestreview-1260834278. The change will make it into next week's release

**Comment by gustavo-delfosim** [comment]
@sryza indeed it is the same bug as reported and fixed in #11738 . Thanks for pointing it out, I will be waiting for the release.
